crates/core/src/operations/writer.rs

```diff
@@ -427,10 +419,15 @@ impl PartitionWriter {
         Ok(())
     }
 
+    /// Retrieves the list of [Add] actions associated with the files written.
+    pub async fn drain_files_written(&mut self) -> Vec<Add> {
```

Comment:
Is the goal for this function to allow for data files (`.parquet`) to be written without the process committing to the Delta table?  I have read the discourse with @hntd187 but I'm still struggling to understand how this public API would be used

Reply:
So let's say in an event where a server has written a lot of data into the table but hasn't committed the additional files yet and there was a power outage. The files will be in storage but not in the metadata. So this can be used to prevent data loss (While writing data, also creating commits of the newly added files).

Reply:
Would something be calling drain periodically while writes are still happening then? What would be the difference vs closing the writer periodically and creating a new one?

Reply:
Saved cost of reallocation, re-initialization and if you are closing after a certain time frame it is possible to create just a lot of small files which aren't optimized. Then will need to OPTIMIZE.

Reply:
Yes, I made an alternative to delta-kafka-ingest for my specific use case and it will be receiving data all day.